cpp/src/io/parquet/page_hdr.cu

```diff
@@ -596,26 +609,43 @@ CUDF_KERNEL void __launch_bounds__(128)
   }
 }
 
-void __host__ DecodePageHeaders(ColumnChunkDesc* chunks,
+void launch_decode_page_headers(ColumnChunkDesc* chunks,
                                 chunk_page_info* chunk_pages,
                                 int32_t num_chunks,
                                 kernel_error::pointer error_code,
                                 rmm::cuda_stream_view stream)
 {
-  dim3 dim_block(128, 1);
-  dim3 dim_grid((num_chunks + 3) >> 2, 1);  // 1 chunk per warp, 4 warps per block
+  static_assert(decode_page_headers_block_size % cudf::detail::warp_size == 0,
+                "Block size for decode page headers kernel must be a multiple of warp size");
+
+  auto constexpr num_warps_per_block = decode_page_headers_block_size / cudf::detail::warp_size;
+
+  auto const num_blocks = cudf::util::div_rounding_up_safe(
+    num_chunks, num_warps_per_block);  // 1 warp per chunk, 4 warps per block
+  dim3 dim_block(decode_page_headers_block_size, 1);
+  dim3 dim_grid(num_blocks,
+                1);  // 1 chunk per warp, 4 warps per block
```

Comment:
same here, "4" seems out of place with the other magic numbers replaced
IMO these comments are not needed